pkg/apis/flows/v1/sequence_validation.go

```diff
@@ -41,15 +41,14 @@ func (ps *SequenceSpec) Validate(ctx context.Context) *apis.FieldError {
 
 	if ps.ChannelTemplate == nil {
 		errs = errs.Also(apis.ErrMissingField("channelTemplate"))
-		return errs
-	}
-
-	if len(ps.ChannelTemplate.APIVersion) == 0 {
-		errs = errs.Also(apis.ErrMissingField("channelTemplate.apiVersion"))
-	}
+	} else {
+		if len(ps.ChannelTemplate.APIVersion) == 0 {
+			errs = errs.Also(apis.ErrMissingField("channelTemplate.apiVersion"))
+		}
 
-	if len(ps.ChannelTemplate.Kind) == 0 {
-		errs = errs.Also(apis.ErrMissingField("channelTemplate.kind"))
+		if len(ps.ChannelTemplate.Kind) == 0 {
+			errs = errs.Also(apis.ErrMissingField("channelTemplate.kind"))
+		}
```

Comment:
There is a function to validate the channel template we can reuse that one: https://github.com/knative/eventing/blob/46407e8814dbc67149be3a1be3d8630b3ab31e69/pkg/apis/messaging/v1/channel_validation.go#L55-L64